### PR TITLE
ZTS: Fix io_uring support check

### PR DESCRIPTION
**Pull request by behlendorf**
### Motivation and Context

Issue #13648

### Description

Not all Linux distribution kernels enable io_uring support by
default.  Update the run time check to verify not only that the
kernel is new enough but that it was built with CONFIG_IO_URING=y.
The `fio --ioengine=io_uring --parse-only` check was intended for
this same purpose but would always incorrectly succeed.

### How Has This Been Tested?

Verified correct behavior on Fedora 35 kernel where CONFIG_IO_URING
is enabled by default, and on AlmaLinux 9 where it is not. 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by tonynguien — APPROVED** [review]

**Review by tonyhutter — APPROVED** [review]

**Review by gmelikov — APPROVED** [review]
LGTM with one minor

**Comment by behlendorf** [comment]
Refreshed to put the `fio` command check back since it is useful to check that a newest version of `fio` is installed.

File: tests/zfs-tests/tests/functional/io/io_uring.ksh [hunk 1]
```diff
@@ -44,6 +44,10 @@ if [[ $(linux_version) -lt $(linux_version "5.1") ]]; then
 	log_unsupported "Requires io_uring support"
```
Comment (gmelikov) [inline tests/zfs-tests/tests/functional/io/io_uring.ksh:44]: Looks like we can remove this check now, no?

Comment (behlendorf) [reply tests/zfs-tests/tests/functional/io/io_uring.ksh]: Good point.  Done.

**Pull request merged**